File: app/src/main/java/com/tronprotocol/app/rag/RetrievalResult.java
package com.tronprotocol.app.rag;

/**
 * Result from a RAG retrieval query
 * 
 * Combines landseek and ToolNeuron retrieval patterns
 */
public class RetrievalResult {
    private TextChunk chunk;
    private float score;
    private RetrievalStrategy strategy;
    
    public RetrievalResult(TextChunk chunk, float score, RetrievalStrategy strategy) {
        this.chunk = chunk;
        this.score = score;
        this.strategy = strategy;
    }
    
    public TextChunk getChunk() {
        return chunk;
    }
    
    public float getScore() {
        return score;
    }
    
    public RetrievalStrategy getStrategy() {
        return strategy;
    }
    
    @Override
    public String toString() {
        return "RetrievalResult{" +
                "score=" + String.format("%.3f", score) +
                ", strategy=" + strategy +
                ", chunk=" + chunk +
                '}';
    }
}

File: app/src/main/java/com/tronprotocol/app/rag/RetrievalResult.kt
package com.tronprotocol.app.rag

/**
 * Result from a RAG retrieval query
 *
 * Combines landseek and ToolNeuron retrieval patterns
 */
data class RetrievalResult(
    val chunk: TextChunk,
    val score: Float,
    val strategy: RetrievalStrategy,
    val strategyId: String = strategy.name,
    val scoreDistribution: ScoreDistribution? = null,
    val stageSource: String? = null
) {
    override fun toString(): String =
        "RetrievalResult{" +
                "score=" + "%.3f".format(score) +
                ", strategy=" + strategy +
                ", strategyId=" + strategyId +
                ", stageSource=" + stageSource +
                ", chunk=" + chunk +
                '}'
}

data class ScoreDistribution(
    val min: Float,
    val max: Float,
    val mean: Float,
    val stdDev: Float
)
